src/EditorFeatures/CSharpTest/CodeActions/InitializeParameter/AddParameterCheckTests.cs

```diff
@@ -1085,6 +1085,204 @@ private void Init()
             }.RunAsync();
         }
 
+        [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/71636")]
+        public async Task TestUpdateExpressionBody4()
+        {
+            await VerifyCS.VerifyRefactoringAsync(
+                """
+                using System;
+
+                class C
+                {
+                    int M(int v, [||]string s) => v switch
+                    {
+                        _ => 0
+                    };
+                }
+                """,
+                """
+                using System;
+                
+                class C
+                {
+                    int M(int v, string s)
+                    {
+                        if (s is null)
+                        {
+                            throw new ArgumentNullException(nameof(s));
+                        }
+
+                        return v switch
+                        {
+                            _ => 0
+                        };
+                    }
+                }
+                """);
+        }
+
+        [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/71636")]
+        public async Task TestUpdateExpressionBody5()
+        {
+            await VerifyCS.VerifyRefactoringAsync(
+                """
+                using System;
+
+                class C
+                {
+                    int M(bool v, [||]string s) => v ? 0 : 1;
+                }
+                """,
+                """
+                using System;
+
+                class C
+                {
+                    int M(bool v, string s)
+                    {
+                        if (s is null)
+                        {
+                            throw new ArgumentNullException(nameof(s));
+                        }
+                
+                        return v ? 0 : 1;
+                    }
+                }
+                """);
+        }
+
+        [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/71636")]
+        public async Task TestUpdateExpressionBody6()
+        {
+            await VerifyCS.VerifyRefactoringAsync(
+                """
+                using System;
+
+                class C
+                {
+                    int M(bool v, [||]string s) => v ?
+                        0 :
+                        1;
+                }
+                """,
+                """
+                using System;
+
+                class C
+                {
+                    int M(bool v, string s)
+                    {
+                        if (s is null)
+                        {
+                            throw new ArgumentNullException(nameof(s));
+                        }
+                
+                        return v ? 0 : 1;
```

Comment:
so this seems undesirable fwiw.

Reply:
@sharwell thoughts on this outcome?  we could let this go, and treat it as an unfortunate consequence of this strategy.  or we could only take a pr here if it didn't undo the user's desired formatting in a case like this.

Reply:
I believe this is just a side effect of the strategy used here. Once we implement a proper re-indenter that we can use in general refactorings, we could migrate from elastic trivia over to that new service and this issue would resolve itself.